### PR TITLE
mxnet: 1.6.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upstream update.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124329` at 8e0a186f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.optuna</li>
    <li>python39Packages.optuna</li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.mxnet</li>
    <li>python39Packages.mxnet</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mxnet</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `bash`.
Near pkgs/applications/science/math/mxnet/default.nix:5:3:
```
  |
5 | , bash
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/applications/science/math/mxnet/default.nix:58:0:
```
   |
58 |     description = "Lightweight, Portable, Flexible Distributed/Mobile Deep Learning with Dynamic, Mutation-aware Dataflow Dep Scheduler";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/applications/science/math/mxnet/default.nix [hunk 1]
```diff
@@ -1,50 +1,63 @@
...
-    ++ lib.optional (!cudnnSupport) "-DUSE_CUDNN=OFF";
-
+  # Build against the system version of OpenMP.
   postPatch = ''
-    substituteInPlace 3rdparty/mkldnn/tests/CMakeLists.txt \
-      --replace "/bin/bash" "${bash}/bin/bash"
-
-    # Build against the system version of OpenMP.
-    # https://github.com/apache/incubator-mxnet/pull/12160
     rm -rf 3rdparty/openmp
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/mxnet/default.nix:33]:
```suggestion
    rm -r 3rdparty/openmp
```

File: pkgs/applications/science/math/mxnet/default.nix [hunk 2]
```diff
@@ -1,50 +1,63 @@
...
+  ] else [ "-DUSE_CUDA=OFF" ])
+  ++ lib.optional (!cudnnSupport) "-DUSE_CUDNN=OFF";
+
   postInstall = ''
     rm "$out"/lib/*.a
   '';
 
   meta = with lib; {
-    description = "Lightweight, Portable, Flexible Distributed/Mobile Deep Learning with Dynamic, Mutation-aware Dataflow Dep Scheduler";
+    description = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/mxnet/default.nix:57]: description shouldn't be multiline. Please use longDescription instead.

```suggestion
    longDescription = ''
```

File: pkgs/applications/science/math/mxnet/default.nix [hunk 3]
```diff
@@ -1,50 +1,63 @@
...
   postPatch = ''
-    substituteInPlace 3rdparty/mkldnn/tests/CMakeLists.txt \
-      --replace "/bin/bash" "${bash}/bin/bash"
-
-    # Build against the system version of OpenMP.
-    # https://github.com/apache/incubator-mxnet/pull/12160
     rm -rf 3rdparty/openmp
   '';
 
+  nativeBuildInputs = [ cmake gtest perl ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/mxnet/default.nix:36]: You are sure gtest does not belong into checkInputs?

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 124329` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.optuna</li>
    <li>python39Packages.optuna</li>
  </ul>
</details>

**Comment by peterhoeg** [comment]
We already have 1.8.0 in master.

**Pull request closed without merging**